src/api/logger-api.ts

```diff
@@ -212,30 +211,23 @@ export const Event = {
  *         parameters. The guideline to create this JSON block
  *         that describes the event is provided against each
  *         enum listed in the Event enum above.
- *
- *  @typedef {{
- *    name: string,
- *    active: boolean,
- *    data: ?JsonObject,
- * }}
  */
-export let PublisherEvent;
+export interface PublisherEvent {
+  name: string;
+  active: boolean;
+  data: unknown;
+}
 
-/* eslint-disable @typescript-eslint/no-unused-vars */
-/**
- * @interface
- */
-export class LoggerApi {
+export interface LoggerApi {
   /**
    * Send a buy-flow event that occurred on the publisher's site to Google.  The
    * ultimate destination is controlled by configuration settings.  Publisher
    * configuration available:
    *   enablePropensity - Sends data to the Propensity to Subscribe ads server.
    *   enableSwgAnalytics - Sends data to Google's analytics server for buy-flow
    *     comparison purposes.
-   * @param {!PublisherEvent} userEvent
    */
-  sendEvent(userEvent) {}
+  sendEvent(_userEvent: PublisherEvent): void;
```

Comment:
Why does _userEvent have an underscore prefix in this interface? Same question for the other interfaces.

Reply:
This is the standard way to mark a parameter as unused in TypeScript

Reply:
Interface parameters should not need to be marked unused, there is no implementation so all parameters are unused.

Reply:
Good catch, I just removed the leading `_`s